drivers/adc/adc_ite_it8xxx2.c

```diff
@@ -135,6 +126,91 @@ static void adc_disable_measurement(void)
 	irq_disable(DT_INST_IRQN(0));
 }
 
+/* Get result for each ADC selected channel. */
+static void adc_it8xxx2_get_sample(const struct device *dev)
+{
+	struct adc_it8xxx2_data *data = DEV_DATA(dev);
+	struct adc_it8xxx2_regs *const adc_regs = ADC_IT8XXX2_REG_BASE;
+
+	if (adc_regs->VCH0CTL & IT8XXX2_ADC_DATVAL) {
+		/* Read adc raw data of msb and lsb */
+		*data->buffer++ = adc_regs->VCH0DATM << 8 | adc_regs->VCH0DATL;
+	} else {
+		LOG_WRN("ADC failed to read (regs=%x, ch=%d)",
+			adc_regs->ADCDVSTS, data->ch);
+	}
+
+	adc_disable_measurement();
+}
+
+static void adc_poll_valid_data(void)
+{
+	struct gctrl_it8xxx2_regs *const gctrl_base =
+		(struct gctrl_it8xxx2_regs *)DT_REG_ADDR(DT_NODELABEL(gctrl));
+	struct adc_it8xxx2_regs *const adc_regs = ADC_IT8XXX2_REG_BASE;
+	const struct device *dev = DEVICE_DT_INST_GET(0);
+	int valid = 0;
+
+	/*
+	 * If the polling waits for a valid data longer than
+	 * the sampling time limit, the program will return.
+	 */
+	for (int i = 0U; i < (IT8XXX2_ADC_SAMPLE_TIME_US /
+	     IT8XXX2_WAIT_NEXT_CLOCK_TIME_US); i++) {
+		/* delay ~15.25us */
+		gctrl_base->GCTRL_WNCKR = 0;
```

Comment:
Can you use `k_busy_wait()` here instead?  I see the ITE flash driver makes 2 consecutive writes to the register to guarantee a minimum delay of 15 us.  `k_busy_wait()` is clearer to the user what you are doing.

Reply:
Done.